src/forder.c

```diff
@@ -801,6 +800,145 @@ SEXP forder(SEXP DT, SEXP by, SEXP retGrpArg, SEXP sortGroupsArg, SEXP ascArg, S
   return ans;
 }
 
+// all(x==1L)
+static bool all1(SEXP x) {
+  if (!isInteger(x))
+    error("internal error: all1 got non-integer"); // # nocov
+  int *xp = INTEGER(x);
+  for (int i=0; i<LENGTH(x); ++i) {
+    if (xp[i] != 1)
+      return false;
+  }
+  return true;
+}
+
+// identical(cols, head(chmatch(key(x), names(x)), length(cols)))
+bool colsKeyHead(SEXP x, SEXP cols) {
```

Comment:
We have multiple places where we use this functionality in R, once this PR merged we should followup with a R wrapper and use it wherever we currently use `identical(cols, head(chmatch(......)))`.

Reply:
Follow-up issue: #6261

Reply:
Follow-up issue: #6262